src/libraries/System.Net.Http/src/System/Net/Http/HttpTelemetry.cs

```diff
@@ -0,0 +1,152 @@
+        {
+            Interlocked.Increment(ref _startedRequests);
+            WriteEvent(1, scheme, host, port, pathAndQuery, versionMajor, versionMinor);
+        }
+
+        [Event(2, Level = EventLevel.Informational)]
+        public void RequestStop()
+        {
+            Interlocked.Increment(ref _stoppedRequests);
+            WriteEvent(2);
+        }
+
+        [Event(3, Level = EventLevel.Error)]
+        public void RequestAborted()
+        {
+            Interlocked.Increment(ref _abortedRequests);
+            WriteEvent(3);
+        }
+
+        protected override void OnEventCommand(EventCommandEventArgs command)
+        {
+            if (command.Command == EventCommand.Enable)
+            {
+                // This is the convention for initializing counters in the RuntimeEventSource (lazily on the first enable command).
+                // They aren't disabled afterwards...
+
+                // The cumulative number of HTTP requests started since the process started.
+                _startedRequestsCounter ??= new PollingCounter("requests-started", this, () => Interlocked.Read(ref _startedRequests))
+                {
+                    DisplayName = "Requests Started",
+                };
+
+                // The number of HTTP requests started per second since the process started.
+                _startedRequestsPerSecondCounter ??= new IncrementingPollingCounter("requests-started-rate", this, () => Interlocked.Read(ref _startedRequests))
+                {
+                    DisplayName = "Requests Started Rate",
+                    DisplayRateTimeScale = TimeSpan.FromSeconds(1)
+                };
+
+                // The cumulative number of HTTP requests aborted since the process started.
+                // Aborted means that an exception occurred during the handler's Send(Async) call as a result of a
+                // connection related error, timeout, or explicitly cancelled.
+                _abortedRequestsCounter ??= new PollingCounter("requests-aborted", this, () => Interlocked.Read(ref _abortedRequests))
+                {
+                    DisplayName = "Requests Aborted"
+                };
+
+                // The number of HTTP requests aborted per second since the process started.
+                _abortedRequestsPerSecondCounter ??= new IncrementingPollingCounter("requests-aborted-rate", this, () => Interlocked.Read(ref _abortedRequests))
+                {
+                    DisplayName = "Requests Aborted Rate",
+                    DisplayRateTimeScale = TimeSpan.FromSeconds(1)
+                };
+
+                // The current number of active HTTP requests that have started but not yet completed or aborted.
+                _currentRequestsCounter ??= new PollingCounter("current-requests", this, () => Interlocked.Read(ref _startedRequests) - Interlocked.Read(ref _stoppedRequests))
```

Comment:
```suggestion
                _currentRequestsCounter ??= new PollingCounter("current-requests", this, () => -Interlocked.Read(ref _stoppedRequests) + Interlocked.Read(ref _startedRequests))
```

`-stopped + started` could be used instead of `started - stopped` to avoid avoid returning a negative value if `_stoppedRequests` is incremented after reading `_startedRequests`

Reply:
It's not possible for _stoppedRequests to be greater than _startedRequests. In most cases with active requests, _stoppedRequests will always be less than _startedRequests. And when all http requests finish completely (response body fully read or it aborts) _stoppedRequests will be equal to _startedRequests.

So, I don't see the value of this change.

Reply:
I suspect @MihaZupan was noticing that there is a race condition because the read operations for _startedRequests and _stoppedRequests occur at different times. In particular imagine starting with _startedRequests = _stoppedRequests = 0 and then this order of execution:
**Counter polling thread:** Read _startedRequests = 0
**Http Event processing thread:** Complete http request is processed _startedRequests = _stoppedRequests = 1
**Counter polling thread:** Read _stoppedRequests = 1
**Counter polling thread:** Compute 0 - 1 = -1 current requests

Reply:
Thx. I see now.

Doing the math as:

`-Interlocked.Read(ref _stoppedRequests) + Interlocked.Read(ref _startedRequests)`

would end up as:

Counter polling thread: Read _stoppedRequests = 0
Http Event processing thread: Complete http request is processed _startedRequests = _stoppedRequests = 1
Counter polling thread: Read _startedRequests = 1
Counter polling thread: Compute -0 + 1 = 1 current requests

Still not perfectly accurate to transitory race conditions but at least it is a positive number.

I'll make this change.